packages/pigeon/lib/dart_generator.dart

```diff
@@ -736,6 +565,238 @@ PlatformException _createConnectionError(String channelName) {
 \t);
 }''');
   }
+
+  void _writeHostMethod(
+    Indent indent, {
+    required String name,
+    required Iterable<Parameter> parameters,
+    required TypeDeclaration returnType,
+    required List<String> documentationComments,
+    required String channelName,
```

Comment:
I'm fine with either. I did it this way since `ProxyApi`s would need to convert its `Field`s and `Constructor`s into a `Method` to use the helper method.  I don't have a preference for either option. I'll convert the helper methods to take a `Method`.

Reply:
If there are multiple uses and they don't all use `Method` then it's fine as is.